analyzers/tests/SonarAnalyzer.UnitTest/TestCases/DisposableNotDisposed.CSharp8.cs

```diff
@@ -0,0 +1,138 @@
+            {
+                // do nothing
+            }
+
+            FileStream fs6;
+            await using ((fs6 = File.Open(@"c:\foo.txt", FileMode.Open)).ConfigureAwait(false))
+            {
+                // do nothing
+            }
+
+            FileStream fs7 = new FileStream(@"c:\foo.txt", FileMode.Open);
+            await using (var ignored = fs7.ConfigureAwait(false));
+
+            FileStream fs8 = new FileStream(@"c:\foo.txt", FileMode.Open);
+            await using var ignored2 = fs8.ConfigureAwait(false);
+
+            using var fs9 = new FileStream(@"c:\foo.txt", FileMode.Open);
+
+            await using var fs10 = new FileStream(@"c:\foo.txt", FileMode.Open);
+
+            await using var fs11 = File.Open(@"c:\foo.txt", FileMode.Open);
+
+            var fs12 = new FileStream(@"c:\foo.txt", FileMode.Open);                     // Compliant - asynchronously disposed manually
+            await fs12.DisposeAsync();
+        }
+
+        public async Task SomePublicAsyncMethod()
+        {
+            await field_fs1.DisposeAsync().ConfigureAwait(false);            
+            await field_fs2.DisposeAsync();
+        }
+
+        private async Task SomePrivateAsyncMethod()
+        {
+            await field_fs3.DisposeAsync();
+        }
+
+        public async ValueTask SomePublicAsyncMethodWithValueTask()
+        {
+            await field_fs4.DisposeAsync();
+        }
+
+        public ValueTask SomePublicMethodWithValueTask()
+        {
+            return field_fs5.DisposeAsync();
+        }
+
+        public ValueTask AnotherPublicValueTaskMethod() => field_fs6.DisposeAsync();
+    }
+
+    public sealed class ImplementsAsyncDisposable : IAsyncDisposable
+    {
+        private readonly FileStream stream;
+
+        public ImplementsAsyncDisposable()
+        {
+            stream = new FileStream(@"c:\foo.txt", FileMode.Open);                      // Compliant - see GitHub issue: https://github.com/SonarSource/sonar-dotnet/issues/5879
+        }
+
+        public async ValueTask DisposeAsync()
+        {
+            await stream.DisposeAsync();
+        }
+    }
+
+    public class FluentAssertionsTest
+    {
+        public void FluentAssertionTypes()
+        {
+            var scope = new AssertionScope();                           // Noncompliant
+            var s = new FluentAssertions.Execution.AssertionScope();    // Noncompliant
+
+            using var _ = new AssertionScope();
+            using (var disposed = new AssertionScope())
+            {
+            }
+        }
+    }
+
+    public ref struct Struct
+    {
+        public void Dispose()
+        {
+        }
+    }
+
+    public class Consumer
+    {
+        public void Method()
+        {
+            using var x = new Struct();
+            var y = new Struct();                                                       // Noncompliant
```

Comment:
nitpick? it doesn't seem aligned with anything

```suggestion
            var y = new Struct();  // Noncompliant
```

Reply:
It's aligned with the rest of the Compliant / Noncompliant comments. I put all of them in the same column for readability.
Should I move these comments in the file next to the code itself in every line?